packages/@aws-cdk/cfnspec/build-tools/build.ts

```diff
@@ -29,9 +40,53 @@ async function main() {
 
   spec.Fingerprint = md5(JSON.stringify(normalize(spec)));
 
-  const outDir = path.join(process.cwd(), 'spec');
-  await fs.mkdirp(outDir);
-  await fs.writeJson(path.join(outDir, 'specification.json'), spec, { spaces: 2 });
+  await fs.mkdirp(path.dirname(outFile));
+  await fs.writeJson(outFile, spec, { spaces: 2 });
+  // eslint-disable-next-line no-console
+  console.log(outFile);
+}
+
+/**
+ * Generate Cfnlint spec annotations from sources and patches
+ */
+async function mergeSpecificationFromDirs(inputDir: string, outFile: string) {
+  const spec: any = {};
+
+  for (const child of await fs.readdir(inputDir)) {
+    const fullPath = path.join(inputDir, child);
+    if (!(await fs.stat(fullPath)).isDirectory()) { continue; }
+
+    const subspec = await loadMergedSpec(fullPath);
+    spec[child] = subspec;
+  }
+
+  await fs.mkdirp(path.dirname(outFile));
+  await fs.writeJson(outFile, spec, { spaces: 2 });
+  // eslint-disable-next-line no-console
+  console.log(outFile);
+}
+
+/**
+ * Load all files in the given directory, merge them and apply patches in the order found
+ *
+ * The base structure is always an empty object
+ */
+async function loadMergedSpec(inputDir: string) {
+  const structure: any = {};
+
+  const files = await fs.readdir(inputDir);
+  for (const file of files.filter(n => n.endsWith('.json')).sort()) {
+    const data = await fs.readJson(path.join(inputDir, file));
+    if (file.indexOf('patch') === -1) {
+      // Copy properties from current object into structure, adding/overwriting whatever is found
+      Object.assign(structure, data);
+    } else {
+      // Apply the loaded file as a patch onto the current structure
+      patch(structure, data);
```

Comment:
minor clarification - this the same patching mechanism as the cfn spec, yeah?

Reply:
Yes.